File: Array & String/Rotate Array/RotateArray.go
func rotate(nums []int, k int)  {
    n := len(nums)
	k %= n
	for x := 0; x < k; x++ {
		last := nums[n-1]
		copy(nums[1:], nums[:n-1])
		nums[0] = last
	}
}

File: Array & String/Rotate Array/RotateArray.rs
impl Solution {
    pub fn rotate(nums: &mut Vec<i32>, k: i32) {
        let k = k as usize % nums.len();
        let mut rotated = vec![0; nums.len()];
        for i in 0..nums.len() {
            rotated[(i + k) % nums.len()] = nums[i];
        }
        nums.clone_from_slice(&rotated);
    }
}